### PR TITLE
prefer chainId over networkId in most cases

### LINKED ISSUE #7963: Prefer chainId over networkId whenever possible
Our stack uses `networkId` from `net_version` extensively. We should switch to using the `chainId` from `eth_chainId` ([EIP here](https://eips.ethereum.org/EIPS/eip-695)) wherever possible. Places/cases include:
- The transaction controller
  - Especially when signing transactions (as suggested by Truffle)
- The UI

### LINKED ISSUE #8668: Treat custom RPC networks with mainnet chainId as mainnet
If a user adds a custom RPC network `chainId` `1` or `0x1`, we should indicate that it is a Mainnet network. This includes:
- Using mainnet contracts, services, balance checkers etc.

### PR DESCRIPTION
**Pull request by brad-decker**
#### Bug Fixes
1. fixes #7963 
2. fixes #8668

### CODE REVIEW

**Comment by darkwing** [comment]
When we say "in most cases", what does that represent?  We might want to document that.

**Comment by brad-decker** [comment]
@darkwing https://github.com/MetaMask/metamask-extension/pull/10594/files#diff-ad1f68882a19fa4247caaa355b64dd6a6ae6c4e3c67b51c5a23e4727a0c91c01R86-R93


I added *some* inline documentation to the selector for networkId, but it might not be sufficient. I'll update the PR description to reflect this as well, but did you have some thoughts on where this documentation should live?

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -330,12 +329,24 @@ export default class MetamaskController extends EventEmitter {
...
-            customVariables: { errorMessage: txMeta.simulationFails?.reason },
-          });
+          this.metaMetricsController.trackEvent(
+            {
+              category: 'Background',
+              properties: {
+                action: 'Transactions',
+                errorMessage: txMeta.simulationFails?.reason,
+                numberOfTokens: metamaskState.tokens?.length ?? 0,
+                numberOfAccounts: Object.keys(metamaskState.accounts ?? {})
```
Comment (ryanml) [inline app/scripts/metamask-controller.js:342]: Same as above with `accounts` and `{}`

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -330,12 +329,24 @@ export default class MetamaskController extends EventEmitter {
...
-            name: 'On Chain Failure',
-            customVariables: { errorMessage: txMeta.simulationFails?.reason },
-          });
+          this.metaMetricsController.trackEvent(
+            {
+              category: 'Background',
+              properties: {
+                action: 'Transactions',
+                errorMessage: txMeta.simulationFails?.reason,
+                numberOfTokens: metamaskState.tokens?.length ?? 0,
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply app/scripts/metamask-controller.js:341]: background on this change: I saw that we had a 'getBackgroundMetaMetrics' selector in the UI folder that is only called from the background and I wanted to kill it with fire. I copied the intent of the selector 'getNumberOfTokens' from the frontend here which checks if tokens is specified.

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (brad-decker) [reply app/scripts/metamask-controller.js:342]: Same as above, but here I did add something that didn't exist before. I'll check into both of these and simplify the selectors if they can be.

**Comment by brad-decker** [comment]
I added a commit that changes the name of `getCurrentNetworkId` to `deprecatedGetCurrentNetworkId`

image

between this and the @deprecated flag, I feel like there is a strong signal to not use this.

**Review by DONALDSULLENDER — APPROVED** [review]

**Comment by brad-decker** [comment]
@ryanml ready for another when you can.

**Review by ryanml — APPROVED** [review]

**Pull request merged**